tests/testTriggers.go

```diff
@@ -10962,6 +10976,237 @@ func TriggerResetDiscardMounts(contexts *[]*scheduler.Context, recordChan *chan
 	})
 }
 
+func TriggerScaleFADAVolumeAttach(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer endLongevityTest()
+	startLongevityTest(ScaleFADAVolumeAttach)
+	defer ginkgo.GinkgoRecover()
+	var wg sync.WaitGroup
+
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: ScaleFADAVolumeAttach,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+
+	setMetrics(*event)
+
+	stepLog := "Adding FADA volumes at scale to validate FADA volumes attachment in scale "
+
+	Step(stepLog, func() {
+		log.InfoD(stepLog)
+		var param = make(map[string]string)
+		var appContexts []*scheduler.Context
+
+		fadaScName := PureBlockStorageClass + time.Now().Format("01-02-15h04m05s")
+		log.Infof("Creating pure_block storage class class: %s", fadaScName)
+		param[PureBackend] = k8s.PureBlock
+		_, err := createPureStorageClass(fadaScName, param)
+		if err != nil {
+			log.Errorf("StorageClass creation failed for SC: %s", fadaScName)
+			UpdateOutcome(event, err)
+		}
+		log.InfoD("Deployng FADA based applications")
+		startTime := time.Now()
+		for x := 0; x < deploymentCount; x++ {
+			pvcName := fmt.Sprintf("%s-%d", pvcNamePrefix, x)
+			namespace := fmt.Sprintf("%s-%d", fadaNamespacePrefix, x)
+			deploymentName := fmt.Sprintf("%s-%d", fadaScName, x)
+			wg.Add(1)
+			go deployFadaApps(fadaScName, pvcName, namespace, deploymentName, &wg, &appContexts, event)
+			if x%10 == 0 {
+				// wait for previous threads to be completed
+				wg.Wait()
+			}
+		}
+		log.InfoD("Validating applications context")
+		ValidateApplications(appContexts)
+		if time.Since(startTime) > podAttachTimeout {
+			UpdateOutcome(event, fmt.Errorf("failed to complete all fada pods attachment in [%v]", podAttachTimeout))
+		}
+
+		stepLog = "Cleaning up the FADA deployments"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for x := 0; x < deploymentCount; x++ {
+				wg.Add(1)
+				go cleanupDeployment(appContexts[x], &wg, event)
+				if x%10 == 0 {
+					// wait for previous threads to be completed
+					wg.Wait()
+				}
+			}
+			log.InfoD("Successfully cleaned up the FADA deployments")
```

Comment:
How about?

```go
sem := make(chan struct{}, 10)
for x := 0; x < deploymentCount; x++ {
	sem <- struct{}{}
	wg.Add(1)
	go func(ctx *appContext) {
		defer wg.Done()
		cleanupDeployment(ctx, &wg, event)
		<-sem
	}(appContexts[x])
}
wg.Wait()
log.InfoD("Successfully cleaned up the FADA deployments")
```

The code uses a semaphore to limit concurrency, ensuring exactly 10 goroutines to run simultaneously, without unnecessary waits between batches. Also, to avoid data races, it passes appContexts[x] directly into the goroutine instead of using x directly.

Reply:
Yes good suggestion, will incorporate it ..